white-paper.md

```diff
@@ -280,7 +280,7 @@ We will apply the same methodology for incentive distribution as the steeped tok
 Versioning and conflicting dependencies are significant challenges, and troubleshooting them can turn into massive time drains.
 To address this, we propose each package be subject to a full dependency scan upon submission so we can ensure that the package complies with the following rules for semantic version ranges.
 
-* Packages may only depend on major versions or greater than specific versions, but for the whole range.
+* Packages may only constrain their dependencies to a major version, though the start of the range can be any valid semantic version (e.g., >=5.2.1 <6).
 * If a dependency is upgraded to a more recent major version, tea may require that the package’s major version be increased.
 * Similarly, if a dependency is upgraded to a more recent minor version, tea may require that the package’s minor version be increased.
```

Comment:
Thinking about these next two bullets a little further, I think it would be interesting to take a package such as [_webpack_](https://github.com/webpack/webpack) (it has 24 first-level dependencies) and see how many times it would've been expected to bump its major version if it followed these rules.

Reply:
That's a very good suggestion.

All the same, this is IMO the correct route for us to take. Just because a dependency bumps its major doesn't mean you (as a consumer) should bump it. Deps should also be encouraged to bump sparingly, and since they will lose token distribution if consumers don’t bump, they should only bump for very good reason.

With tea consumers should group major bumps and deps should bump sparingly so overall the open source ecosystem will become more stable and less subject to the whim of cowboy development.

Reply:
I actually will probably submit a PR for this myself since I think you should be able to accept a range of majors, eg. React 16 and 17 are API compatible.

The *smallest* range you can depend on is a major.

Essentially we are trying to prohibit depending on eg. a minor